### DISCUSSION TITLE
Difference between the node and browser environment

### DISCUSSION THREAD

**Discussion by davidmz** [solidjs/solid#1065] [Q&A] [answered]
Hi, please look at this very simple and meaningless code ([playground](https://playground.solidjs.com/?version=1.4.1#NobwRAdghgtgpmAXGGUCWEwBowBcCeADgsrgM4Ae2YZA9gK4BOAxiWGjIbY7gAQi9mjOFFxwAwrU70xAEyyDhouACVatXAqEixAZTQBzaABteAX14AzRlN4AdGrWNpZAWgBWZBwG47EP5b0EMy4aLQQijpwALJojDaMABQYaKFQxgCU-H68guFkfMBkLApkcLi6LAC6vAC8kcr6RunJEKlo6Rm+EXkQBbzAsnAFpeUAIsO4NfXajYYmiV05DWKS0nKJi3UAfLxluBMFicXMixlLPcK4TBGDk6MV1d1mfn4A9G+8ACqTvKkAhH5ZmI1BpNllarsQMtmPlCjA4gkHrF4txpisYojuIkACwAJguuVhfSccAAdMZaAZEgjUUlzt1cvsUQlcQBmQm9OjGcmU6m01kMvxmC5gMxVIA)). This code use only regular Solid functions, there is no compilation and code transformations.

As you can see, the _createMirror_ simply copies value from _src_ to _dst_ using _createComputed_. And it works well in the playground (i.e.  in browser): the console prints 42 and 43.

But when I run the same code in node command-line, I see the different output: 42 and 42. The _setMirror_ function doesn't re-run the _createComputed_ body.

What is the difference here, and how can I get node to behave like a browser (in a simple way)?

**Comment by atk** [comment]
solid-js/server is meant to provide static server-side-rendered pages, to be hydrated on the client to add the reactivity, so this behavior is to be expected, unless you use an async rendering, e.g. to stream.

If you want the same behaviour as in the current, you need a DOM and a few other things; the easiest way to get those I implemented in solid-register.

**Comment by davidmz** [comment, marked as answer]
Well, thanks @atk for the tip, I figured it out. It's probably common knowledge, but I'll write just in case.

Solid [uses](https://github.com/solidjs/solid/blob/v1.4.4/packages/solid/package.json#L39) the heavy [conditional exports](https://nodejs.org/api/packages.html#conditional-exports) magic, so the `import … from 'solid-js'` in browser and in node will import different functions (although with the same signatures). The node (server-side) version is entirely devoid of reactivity.

Unfortunately, the tests are also run in the node environment, so with a naive run (as I did) we cannot test reactivity. We have the following options here:

1. Run tests through with special adapters (solid-jest etc.) that do their own magic, neutralizing the magic of conditional export;
2. Explicitly import from `solid-js/dist/solid.js` (the browser Solid version) instead of `solid-js`;
3. Create `"solid-js": "solid-js/dist/solid.js"` rule in the `resolve.alias` in the test/build config;
4. Run `node --conditions browser simple-test.js` instead of `node simple-test.js`;
5. ???

I hope this helps someone in the future.

**Reply by atk** [reply]
I already documented most of this in the testing guide and several dev.to posts.
